src/App.tsx

```diff
@@ -1,61 +1,124 @@
+import { FormEvent, useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todoList, setTodoList] = useState(todos);
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [isUserError, setIsUserError] = useState(false);
+
+  const maxId = () => {
+    return Math.max(...todoList.map(todo => todo.id)) + 1;
+  };
+
+  const addTodo = () => {
+    const newTodo = {
+      id: maxId(),
+      title,
+      completed: false,
+      userId,
+      user: getUser(userId),
+    };
+
+    setTodoList(currentTodos => {
+      return [...currentTodos, newTodo];
+    });
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    if (title.length === 0) {
+      setIsTitleError(true);
+    }
```

Comment:
```suggestion
    if (!title) {
      setIsTitleError(true);
    }
```
we can check like this for falsy title